tkg/client/common.go

```diff
@@ -323,9 +322,21 @@ func (c *TkgClient) SetAllowLegacyClusterConfiguration() string {
 		log.V(6).Infof("Setting %v to %q", constants.ConfigVariableAllowLegacyCluster, allowLegacyCluster)
 		c.TKGConfigReaderWriter().Set(constants.ConfigVariableAllowLegacyCluster, allowLegacyCluster)
 	} else {
-		log.V(6).Infof("Info: %v configuration already set to %q", constants.ConfigVariableAllowLegacyCluster, value)
 		// ALLOW_LEGACY_CLUSTER is explicitly set in cluster config file
-		allowLegacyCluster = value
+		log.V(6).Infof("Info: %v configuration already set to %q", constants.ConfigVariableAllowLegacyCluster, value)
+		if value != "true" && value != "false" {
```

Comment:
Can it be defined as a bool variable in the config file?

Reply:
@DanielXiao , yes, it can be defined as a bool variable int he config file. 
The value is a string type is caused by `TKGConfigReaderWriter().Get` func will alway return a string, you can check [here](https://github.com/kubernetes-sigs/cluster-api/blob/main/cmd/clusterctl/client/config/reader_viper.go#L173-L177). 
And `Get` func is calling `GetString` func, [GetString](https://github.com/spf13/viper/blob/master/viper.go#L951-L955) func will alway converts a value to a string type.
```
func GetString(key string) string { return v.GetString(key) }

func (v *Viper) GetString(key string) string {
	return cast.ToString(v.Get(key))
}
```
So I think `ALLOW_LEGACY_CLUSTER`'s  is `true`(bool type) or `"true"`(string type) will cause the same action.